builtInFunctions/saveUserName_test.go

```diff
@@ -10,16 +10,19 @@ import (
 )
 
 func TestNewSaveUserNameFunc(t *testing.T) {
-	m, err := NewSaveUserNameFunc(0, nil, nil)
+	m, err := NewSaveUserNameFunc(0, nil, nil, nil)
 	require.Equal(t, err, ErrNilDnsAddresses)
 
-	m, err = NewSaveUserNameFunc(0, make(map[string]struct{}), nil)
+	m, err = NewSaveUserNameFunc(0, make(map[string]struct{}), nil, nil)
+	require.Equal(t, err, ErrNilDnsAddresses)
+
+	m, err = NewSaveUserNameFunc(0, make(map[string]struct{}), make(map[string]struct{}), nil)
 	require.Equal(t, err, ErrNilEnableEpochsHandler)
 
 	dnsAddr := []byte("DNS")
 	mapDnsAddresses := make(map[string]struct{})
 	mapDnsAddresses[string(dnsAddr)] = struct{}{}
-	m, err = NewSaveUserNameFunc(0, mapDnsAddresses, &mock.EnableEpochsHandlerStub{})
+	m, err = NewSaveUserNameFunc(0, mapDnsAddresses, mapDnsAddresses, &mock.EnableEpochsHandlerStub{})
```

Comment:
Maybe this maps should be different for a better test!

Reply:
done